fenix/app/src/main/java/org/mozilla/fenix/onboarding/controller/OnboardingController.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.controller
+
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.onboarding.interactor.OnboardingInteractor
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * An interface that handles the view manipulation of the first run onboarding.
+ */
+interface OnboardingController {
+    /**
+     * @see [OnboardingInteractor.onStartBrowsingClicked]
+     */
+    fun handleStartBrowsingClicked()
+
+    /**
+     * @see [OnboardingInteractor.onReadPrivacyNoticeClicked]
+     */
+    fun handleReadPrivacyNoticeClicked()
+}
+
+/**
+ * The default implementation of [OnboardingController].
+ */
+class DefaultOnboardingController(
+    private val activity: HomeActivity,
+    private val hideOnboarding: () -> Unit,
+) : OnboardingController {
+
+    override fun handleStartBrowsingClicked() {
+        hideOnboarding()
```

Comment:
Not a blocker for this PR nor follow-up homework for you. I just find it weird that a lambda is being passed in, when this smells like it should probably be some sort of store/state/action response instead.

Reply:
Ya, it is weird that lambda from the fragment is being passed into the controller when ideally we would want to contain this logic inside of the controller and keep the fragment as minimal as possible. This isn't an isolated case. We shouldn't need to have this function when we do decouple the first run onboarding. You will see even in https://bugzilla.mozilla.org/show_bug.cgi?id=1822606 I am just trying to remove logic from the fragment.

Reply:
Sounds good. Looking forward to that PR